lib/multiplexer/multiplexer_test.go

```diff
@@ -721,6 +731,70 @@ func TestMux(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, "db listener", string(dbBytes))
 	})
+
+	t.Run("SSHProxyHelloSignature", func(t *testing.T) {
+		// Ensures SSH connections fronted with the ProxyHelloSignature are
+		// detected as SSH connections.
+		t.Parallel()
+		listener, err := net.Listen("tcp", "127.0.0.1:0")
+		require.NoError(t, err)
+
+		mux, err := New(Config{
+			Listener:            listener,
+			EnableProxyProtocol: true,
+		})
+		require.NoError(t, err)
+		go mux.Serve()
+		defer mux.Close()
+
+		called := false
+		sshHandler := sshutils.NewChanHandlerFunc(func(_ context.Context, _ *sshutils.ConnectionContext, nch ssh.NewChannel) {
+			called = true
+			err := nch.Reject(ssh.Prohibited, "nothing to see here")
+			require.NoError(t, err)
+		})
+
+		srv, err := sshutils.NewServer(
+			"test",
+			utils.NetAddr{AddrNetwork: "tcp", Addr: "localhost:0"},
+			sshHandler,
+			[]ssh.Signer{signer},
+			sshutils.AuthMethods{Password: pass("abc123")},
+		)
+		require.NoError(t, err)
+		go srv.Serve(mux.SSH())
+		defer srv.Close()
+
+		// Manually create client conn so we can inject the ProxyHelloSignature
+		conn, err := net.DialTimeout("tcp", listener.Addr().String(), time.Second)
+		require.NoError(t, err)
+		hp := &apisshutils.HandshakePayload{
+			ClientAddr: "6.6.6.6",
```

Comment:
Should we check that the correct IP is passed along correctly?

Reply:
[1dd218b](https://github.com/gravitational/teleport/pull/14712/commits/1dd218b44407b6e9d4c1b2cebecc08f1361e08f8)